commons/src/main/kotlin/com/github/quillraven/commons/ashley/system/CameraLockSystem.kt

```diff
@@ -0,0 +1,29 @@
+package com.github.quillraven.commons.ashley.system
+
+import com.badlogic.ashley.core.Entity
+import com.badlogic.ashley.systems.IteratingSystem
+import com.badlogic.gdx.graphics.Camera
+import com.github.quillraven.commons.ashley.component.CameraLockComponent
+import com.github.quillraven.commons.ashley.component.TransformComponent
+import com.github.quillraven.commons.ashley.component.transformCmp
+import ktx.ashley.allOf
+import ktx.log.error
+import ktx.log.logger
+
+class CameraLockSystem(private val camera: Camera) :
+    IteratingSystem(allOf(CameraLockComponent::class, TransformComponent::class).get()) {
+    override fun processEntity(entity: Entity, deltaTime: Float) {
+        if (entities.size() > 1) {
+            LOG.error { "There are more than 1 entities with a locked camera" }
+        }
+
+        with(entity.transformCmp.position) {
+            camera.position.set(x, y, 0f)
+        }
+        camera.update()
+    }
+
+    companion object {
+        private val LOG = logger<CameraLockSystem>()
+    }
+}
```

Comment:
You can add an `.editorconfig` to avoid warnings about missing whitespace. See KTX config for an example. It works out of the box in IntelliJ.

Reply:
Where do you see this warning?

Reply:
@Quillraven
![image](https://user-images.githubusercontent.com/11707612/109780379-41790e80-7c07-11eb-9423-efd0e3854225.png)
You can see it when you enter the changes view. `.editorconfig` will fix that, as the newlines will be added automatically by IDE.

Reply:
Thanks for the hint. However, this feature does not seem to work properly for me in Ultimate 2020.3. It seems to make indents, whitespace trimming, etc. but it does not create newlines although I have `insert_final_newline` set to true.

I created an `.editconfig` in the root folder with following content:

```
root = true

[*]
indent_style = space
indent_size = 4
end_of_line = lf
charset = utf-8
trim_trailing_whitespace = true
insert_final_newline = true
```

I then go to the src folder of the commons project and press CTRL + ALT + L, tick "include subdirectories" and click on run. It fixes the formatting of some files but e.g. it does not add the newline to `CameraLockSystem`. Any idea how to fix that?

Reply:
Afaik it works if you actually edit the file. Never had any issues with it, so not sure if I can help. Using built-in IntelliJ settings is also an option, but `.editorconfig` should handle that for other contributors.

Reply:
@Quillraven You've copied the 2 space indent size. It's mostly up to personal preference - this is the setting that I've used in KTX, since I find it easier to read. You might want to revert the last commit and set it to 4 spaces instead if you'd like to avoid as many changes in the project. I think 4 spaces are recommended by JetBrains code guides.

Reply:
I looked up some examples and saw this one: https://github.com/JetBrains/kotlin-playground/blob/master/.editorconfig

If the kotlin playground projects uses 2 spaces and master czyzby does the same then who I am to choose a different setting? :D